src/sentry/tasks/app_store_connect.py

```diff
@@ -80,3 +79,27 @@ def create_difs_from_dsyms_zip(dsyms_zip: str, project: Project) -> None:
         created = debugfile.create_files_from_dif_zip(fp, project, accept_unknown=True)
         for proj_debug_file in created:
             logger.debug("Created %r for project %s", proj_debug_file, project.id)
+
+
+@instrumented_task(
+    name="sentry.tasks.app_store_connect.refresh_all_builds", queue="appstoreconnect"
+)  # type: ignore
+def refresh_all_builds() -> None:
+    # We have no way to query for AppStore Connect symbol sources directly, but
+    # getting all of the project options that have custom symbol sources
+    # configured is a reasonable compromise, as the number of those should be
+    # low enough to traverse every hour.
+    # Another alternative would be to get a list of projects that have had a
+    # previous successful import, as indicated by existing `AppConnectBuild`
+    # objects. But that would miss projects that have a valid AppStore Connect
+    # setup, but have not yet published any kind of build to AppStore.
+    options = ProjectOption.objects.filter(key=appconnect.SYMBOL_SOURCES_PROP_NAME)
+    for option in options:
+        try:
+            all_sources = json.loads(option.value)
+            for source in all_sources:
+                source_id = source.get("id")
+                if source.get("type") == appconnect.SYMBOL_SOURCE_TYPE_NAME and id is not None:
```

Comment:
😱   `id` is a builtin function and thus is never None and neither typechecker nor linter nor manual review caught this.  only realised this once i was editing this function

Reply:
🤦🏻‍♂️ oh well, good thing you caught that!